### DISCUSSION TITLE
Recomendação de PIX em aplicativo Android

### DISCUSSION THREAD

**Discussion by ihasime** [bacen/pix-api#533] [General]
Olá pessoal 

sou novo neste PIX e gostaria de tirar umas dúvidas sobre a melhor abordagem. Estou desenvolvendo uma solução móvel que mostrará um valor para o usuário pagar e este valor depende dos itens que ele escolher mas não são valores altos (abaixo dos R$30). Preciso gerar um PIX com tempo de vida e saber seu status se já foi pago ou não. Qual seria a melhor abordagem: **QR estático ou dinâmico** ? (Tô lendo sobre o dinâmico e **travei neste "location"**)

Como é uma solução a baixo custo gostaria de usar o PIX mas pelo q estou vendo todas as APIs estão cobrando % o que não é ideal para mim. **Gostaria de fugir disso** se possível. 

Até agora consegui gerar o payload para um PIX estático com txid de acordo com o https://www.bcb.gov.br/content/estabilidadefinanceira/pix/Regulamento_Pix/II_ManualdePadroesparaIniciacaodoPix.pdf. Tanto o QR code quanto o payload foram reconhecidos corretamente pelo meu aplicativo do banco (não cheguei a realizar o pagamento mas apareceu para pagar com as infos certinhas). Não cheguei a entender como faço para **consultar pelo txid** ... é possível (li que é possível com txid) ? Alguém poderia me explicar como fez ?

Desde já agradeço qualquer ajuda

Obrigado

**Comment by daniloab** [comment]
Ola @ihasime 

QR estático ou dinâmico é bem tranquilo.

O estatico é um qrcode que nao possui expiração, pode receber N vezes o mesmo valor ou até não possui nenhum valor pré-cadastrado possibilitando assim o pagador inserir um valor que desejar no momento do pagamento.

O dinâmico possui os pontos que citei acima: expiração, pagamento único, entro outros.

Deixa aqui uma dica para você utilizar a https://www.woovi.com. Possui ambos qrcodes com conciliação em tempo real com webhooks.

**Reply by ihasime** [reply]
obg ! vou dar uma olhada :)

**Comment by rubenskuhl** [comment]
Para ticket baixo, a cobrança por % é até melhor. 

Não tem como fugir da API, pois apenas um PSP pode receber Pix e te avisar que foi recebido. E não vi nenhum PSP diferenciando mais preço de QR-Code estático de QR-Code dinâmico.

O que você poderia considerar ao invés de QR-Code é usar OpenFinance com Iniciador de Pagamento; se o seu usuário é mobile, isso dará uma experiência mais fluida do que o copia-e-cola do QR-Code. Você pode ver a descrição de um dos PSPs autorizados a isso em:
https://gerencianet.com.br/open-finance/

A lista de todos os iniciadores de pagamento autorizados é esta:
https://openfinancebrasil.org.br/quem-participa/?marca=&modalidade=PAGTO

**Reply by ihasime** [reply]
hmmm interessante . vou dar uma olhada! obg !

**Comment by ihasime** [comment]
Olá pessoal uma outra dúvida ... preciso receber e enviar PIX e vi que com o gerencianet posso tanto receber (com taxa de 1.19%) e enviar (com taxa "a negociar"). Como minha solução está bem no começo estou tentando evitar muitas taxas (mesmo sendo pequenas). Vi que o openpix.com.br cobra 0.8% mas não se eu posso enviar PIX através de sua API (mandei mensagem, estou aguardando uma resposta).  O openpix.com.br é seguro ? Achei o site woovi.com muito parecido com o do openpix.com.br o que me deixou meio confuso ... são a mesma empresa ?

**Reply by rubenskuhl** [reply]
São a mesma empresa. A OpenPix/Woovi não é um PSP, eles usam outros PSPs (um único de escolha deles no plano Integrado, vários no plano Enterpris). 

Sobre envio de Pix, eu mantenho uma lista de prestadores em #480 , complementar à lista de PSPs com recebimento de Pix em #76 .

**Reply by ihasime** [reply]
obg ! :D
